job/replication/transfer.go

```diff
@@ -192,7 +192,18 @@ enter:
 		return "", err
 	}
 	if !exist {
-		err := c.createProject()
+		project, err := dao.GetProjectByName(c.project)
+		if err != nil {
+			c.logger.Errorf("an error occurred while getting project %s on %s: %v", c.project, c.srcURL, err)
+			return "", err
+		}
+
+		isPublic := false
+		if project != nil {
+			isPublic = project.Public == 1
+		}
+
+		err = c.createProject(isPublic)
```

Comment:
maybe c.createProject(project.Public) is sufficient as project can never be nil otherwise an error will return?